### DISCUSSION TITLE
怎么把ts侧用protobufjs生成的uint8Array 传到C++里并转成void*的数据格式

### DISCUSSION THREAD

**Discussion by nathan77886** [Tencent/puerts#1230] [Q&A]
我现在尝试在ts层接入protobuf
现在ts是这么实现的
` TestSendPb(): void {`
`    const reqTest = encodeC2M_TestRequest({`
`      RpcId: 1,`
`      ActorId: 8888,`
 `     request:"1"`
`    });`
 `   console.log(reqTest);`
`  }`
C++里是这么搞的
`	UFUNCTION()`
`       void SendPlatformMessage(TArray<uint8> buffer);`
`void UGS_ServerClient::SendPlatformMessage(TArray<uint8> Buffer)`
`{`
`	if (!Socket || !this) {`
`		return;`
`	}`
`	auto arrayData = TArray<TArray<uint8>>();`
`	arrayData.Add(Buffer);`
`	if (Buffer.Num() > 0 && Socket->IsConnected()) {`
`		Socket->Send(arrayData.GetData(), Buffer.Num(), true);`
`	}`
`}`
但我感觉我这么搞不对，我的C++功底不行，有没有大佬能指导一下啊

**Comment by chexiongsheng** [comment]
看arraybuffer的例子：https://github.com/chexiongsheng/puerts_unreal_demo/blob/master/TsProj/QuickStart.ts#L107

**Comment by nathan77886** [comment]
大佬，要是把C++里的传回到ts，怎么封格式呢
![image](https://user-images.githubusercontent.com/31683427/223983457-5806d254-0278-40c1-bbb0-9f373499ec58.png)
这么处理吗

**Reply by nathan77886** [reply]
这样子生成貌似会有问题，broadcast 出来的结构体是个空对象

**Reply by chexiongsheng** [reply]
有这种问题应该提issue呀
